src/config/rcxml.c

```diff
@@ -437,13 +477,17 @@ entry(xmlNode *node, char *nodename, char *content)
 		rc.snap_top_maximize = get_bool(content);
 
 	/* <windowSwitcher show="" preview="" outlines="" /> */
-	} else if (!strcasecmp(nodename, "show.windowSwitcher.core")) {
+	} else if (!strcasecmp(nodename, "show.windowSwitcher")) {
 		rc.cycle_view_osd = get_bool(content);
-	} else if (!strcasecmp(nodename, "preview.windowSwitcher.core")) {
+	} else if (!strcasecmp(nodename, "preview.windowSwitcher")) {
 		rc.cycle_preview_contents = get_bool(content);
-	} else if (!strcasecmp(nodename, "outlines.windowSwitcher.core")) {
+	} else if (!strcasecmp(nodename, "outlines.windowSwitcher")) {
 		rc.cycle_preview_outlines = get_bool(content);
 
+	/* Remove this long term - just a friendly warning for now */
+	} else if (strstr(nodename, "windowswitcher.core")) {
+		wlr_log(WLR_ERROR, "<windowSwitcher> should not be child of <core>");
+
```

Comment:
Should we also parse the old variants? e.g. something like
```c
if (strcasestr(nodename, "windowSwitcher.core")) {
	string_truncate_at_pattern(nodename, ".core");
	wlr_log(WLR_ERROR, "<windowSwitcher> should not be child of <core>");
}

..
} else if (!strcasecmp(nodename, "show.windowSwitcher") {
	rc.cycle_view_osd = get_bool(content);
} ..

```

Reply:
Okay, good point. Now supporting old variants.
Spelt out the full nodenames to make it super-clear what's happening rather than using the string_truncate... helper.